### PR TITLE
Updating TOC and adding Assign docs

### LINKED ISSUE #2102: Add docs for Assign method in C# Markup
## What would you like clarification on:

Currently there doesn't seem to be any docs outlining the use of the Assign method to allow us to capture a given control and use it later as we build out the UI or hook up events.

## Concern?

- [ ] Usage in industry
- [x] Clarification of capabilities
- [x] Getting started with Uno
- [ ] Developing with Uno
- [ ] Contributing to the Uno project
- [ ] Publishing your application
- [ ] Support
- [ ] Other (please specify):

### PR DESCRIPTION
**Pull request by dansiegel**
GitHub Issue (If applicable): closes #

- closes #2102

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

### CODE REVIEW

File: doc/toc.yml [hunk 1]
```diff
@@ -67,11 +67,10 @@
         - name: Enable internal Uno Platform logging
           href: Learn/Tutorials/Logging/HowTo-InternalLogging.md
     - name: Markup
-      href: xref:Overview.Markup.Overview
```
Comment (nickrandolph) [inline doc/toc.yml:70]: keep this

File: doc/toc.yml [hunk 2]
```diff
@@ -67,11 +67,10 @@
...
       items:
-        - name: Generating Extensions
-          href: Overview/Markup/GeneratingExtensions.md
-        - name: Dependency Property Builder
+        - name: Overview
+          href: Overview/Markup/Overview.md
```
Comment (nickrandolph) [inline doc/toc.yml:72]:
```suggestion
          href: xref:Overview.Markup.Overview
```

File: doc/toc.yml [hunk 3]
```diff
@@ -67,11 +67,10 @@
         - name: Enable internal Uno Platform logging
           href: Learn/Tutorials/Logging/HowTo-InternalLogging.md
     - name: Markup
-      href: xref:Overview.Markup.Overview
       items:
```
Comment (nickrandolph) [inline doc/toc.yml:70]:
```suggestion
      href: xref:Overview.Markup.Overview
      items:
```

**Review by nickrandolph — APPROVED** [review]

**Comment by dansiegel** [comment]
@backportio backport release/stable/3.0

**Pull request merged**